pkg/subnet/kube/kube.go

```diff
@@ -226,6 +241,36 @@ func newKubeSubnetManager(ctx context.Context, c clientset.Interface, sc *subnet
 	return &ksm, nil
 }
 
+func (ksm *kubeSubnetManager) enqueueLeaseEvent(evt lease.Event, nodeName string) {
+	// Try to send immediately
+	select {
+	case ksm.events <- evt:
+		return
+	default:
+		log.Infof("Channel buffer full, add event asynchronously")
+	}
+
+	// Instead of select with default, *block* until a slot is free
+	ksm.maxAsyncSend <- struct{}{} // acquire a slot, block if none available
```

Comment:
Could use high level library https://pkg.go.dev/golang.org/x/sync/semaphore instead

Reply:
Ack - I was originally trying to avoid adding new ext library dependencies, but makes sense in this case for readability. Updated.